internal/urlutil/url.go

```diff
@@ -149,6 +149,59 @@ func GetDomainsForURL(u *url.URL, includeDefaultPort bool) []string {
 	return []string{u.Hostname(), net.JoinHostPort(u.Hostname(), defaultPort)}
 }
 
+func AllDomainsForURL(u *url.URL, includeDefaultPort bool) func(yield func(string, string) bool) {
```

Comment:
The use of closure functions like this is to avoid allocating a slice?

Reply:
Oh I see this is compatible with v1.23's upcoming `Seq2` type?

Reply:
Yep, this was to avoid duplicating some slices - when go 1.23 is released it can be rewritten as `for host, port := range AllDomainsForUrl(...)` but for now we call it "manually" like `AllDomainsForUrl(func(host, port){ ... })` which has the same effect